api/tasks/http.go

```diff
@@ -26,10 +26,14 @@ func AddTask(w http.ResponseWriter, r *http.Request) {
 	taskObj.Status = "waiting"
 	taskObj.UserID = &user.ID
 
-	if err := db.Mysql.Insert(&taskObj); err != nil {
+	if err := util.UserVaultCache.IncrementUsage(user.Username); err != nil {
 		panic(err)
 	}
 
+	if err := db.Mysql.Insert(&taskObj); err != nil {
+		util.UserVaultCache.DecrementUsage(user.Username)
+		panic(err)
```

Comment:
do we REALLY need to panic here? Semaphore is way too panicky at the moment, and that's very problematic when running in docker. It's on the cards to make it generally less panicky (#462) so i'd prefer not to introduce more. Can we soft fail in some nice way here and just log the error, or at least recover from the panic?

Reply:
The AddTask function does not have any error handling but panicking in every error case. If we can't create a task, we have nothing left to do but to throw an error.
For another application (https://github.com/joernott/go-ampel), I used to reply with the correct http status codes instead of failing. I would recommend to do that in a consistent way in a separate PR for #462 for the whole http.go  instead of just doing it for this one call.

Reply:
Throwing an error and logging it is definitely correct, imo panicking is not (though as previously stated I am aware we do this too often atm), because a panic crashes the entire program, including the web interface, which I do not think should be the case for any error (the be and fe should effectively be totally decoupled even though they are not currently).

I also agree that we should handle this in a consistent way i'm just reluctant to let in more code that we immediately need to change due to panic handling as we are creating technical debt. 
Would you be interested in tackling #462 after this feature is added? That would give me more peace of mind ;)

Reply:
I can't promise how fast I am going to solve #462 that because I am working on this in the evenings or when I have to wait for my co-workers in my current project. So I can't really predict the time I cam spend per day/week, but I will definitively give it a go.

As I am running semaphore inside a docker container (one conforming to my own guidelines), I definitely need less panic and more clever error handling. For the moment, docker swarm takes care of just restarting the container and the application doesn't panic that often, so the impact on my tiny production environment is negligable. However, running it in large scale multi-user environment with hundreds of servers and many concurrent tasks requires a very stable semaphore to be tagged as production ready.

Reply:
^ Absolutely agreed! At work we run it with systemd so it restarts, but I run it for myself in openshift, which although it takes care of the restarts, means that I see a constant huge count of container restarts, and loose my metrics every time etc..... So it would be really good to solve this. I might also take a crack at  #462 so maybe we can collaborate on that together, and as you are a docker user I might give you a mention when I need docker stuff testing if that would be ok? I'm planning to rework the docker containers in the near future anyway so we can have development builds, are not restricted to pulling bundles from github releases etc.... so I would definitely appreciate someone taking a look at these changes when they happen.

Reply:
I am happy to help and work together with you on those two.